doc/how_to/custom_components/reactive_esm/reactive_esm_layout.md

```diff
@@ -0,0 +1,218 @@
+# Create Layouts With ESM Components
+
+In this guide we will show you how to build custom layouts `JSComponent`, `ReactComponent` or `PreactComponent`.
+
+## Layout a single Panel Component
+
+You can layout a single Panel component as follows.
+
+:::::{tab-set}
+
+::::{tab-item} `JSComponent`
+
+```{pyodide}
+import panel as pn
+from panel.custom import JSComponent, Child
+
+pn.extension()
+
+class LayoutSingleObject(JSComponent):
+    object = Child()
+
+    _esm = """
+export function render({ children }) {
+    const containerID = `id-${crypto.randomUUID()}`;;
+    const div = document.createElement("div");
+    div.innerHTML = `
+    <div>
+        <h1>Temperature</h1>
+        <h2>A measurement from the sensor</h2>
+        <div id="${containerID}">...</div>
+    </div>`;
+    const container = div.querySelector(`#${containerID}`);
+    container.appendChild(children.object)
+    return div;
+}
+"""
+
+dial = pn.widgets.Dial(
+    name="°C",
+    value=37,
+    format="{value}",
+    colors=[(0.40, "green"), (1, "red")],
+    bounds=(0, 100),
+)
+js_layout = LayoutSingleObject(
+    object=dial,
+    name="Temperature",
+    styles={"border": "2px solid lightgray"},
+    sizing_mode="stretch_width",
+)
+js_layout.servable()
+```
+
+::::
+
+::::{tab-item} `ReactComponent`
+
+```{pyodide}
+import panel as pn
+
+from panel.custom import Child, ReactComponent
+
+pn.extension()
+
+class LayoutSingleObject(ReactComponent):
+    object = Child()
+
+    _esm = """
+export function render({ children }) {
+    return (
+        <div>
+            <h1>Temperature</h1>
+            <h2>A measurement from the sensor</h2>
+            <div>
+                {children.object}
+            </div>
+        </div>
+    );
+}
+"""
+
+dial = pn.widgets.Dial(
+    name="°C",
+    value=37,
+    format="{value}",
+    colors=[(0.40, "green"), (1, "red")],
+    bounds=(0, 100),
+)
+react_layout = LayoutSingleObject(
+    object=dial,
+    name="Temperature",
+    styles={"border": "2px solid lightgray"},
+    sizing_mode="stretch_width",
+)
+react_layout.servable()
+```
+
+::::
+
+::::{tab-item} `PreactComponent`
+
+```{pyodide}
+import panel as pn
+
+from panel.custom import Child, PreactComponent
```

Comment:
The `PreactExample` no longer works

![image](https://github.com/holoviz/panel/assets/42288570/f211dc0b-aff0-416d-b024-ea98b590168b)